test/tests/functional/pbs_qmgr.py

```diff
@@ -124,3 +125,33 @@ def test_listcmd_whitespaces(self):
             if vndef_file is not None:
                 self.mom.delete_vnodes()
                 self.server.manager(MGR_CMD_DELETE, VNODE, id=nodename)
+
+    def test_multi_attributes(self):
```

Comment:
So, going by the JIRA ticket, this test verifies that scenario 2 (cyclic dependency) gets handled correctly. But, don't we also need a test to verify that scenario 1 (atomicity) gets handled correctly? or do we already have such a test somewhere else?

Reply:
@agrawalravi90 this test covers scenario 1 as well 
line 147 a = {'queue': 'workq', 'partition': 'bar'}

above we are setting multiple attributes. Also verifying that if partition is not set then workq is also unset.